src/molara/gui/mos_dialog.py

```diff
@@ -415,24 +505,654 @@ def calculate_voxelgrid(self) -> None:
+        """
+        voxel_size_min = 0.03
+        voxel_size_max = 0.43
+        a = voxel_size_min - voxel_size_max
+        b = voxel_size_max
+
+        return a * self.ui.isolineResolutionSpinBox.value() + b
+
+    def calculate_isoline_voxelgrid(self) -> None:
+        """Calculate the 2D voxel grid for the isolines."""
+        assert self.aos is not None
+        assert self.mos is not None
+
+        origin = self.isoline_border_origin
+        size = self.isoline_border_size * self.isoline_border_scale
+
+        voxel_size = self.isoline_border_direction * self.isoline_voxel_size_value()
+
+        mo_coefficients = self.mos.coefficients[:, self.selected_orbital]
+
+        voxel_number = np.array(
+            [
+                int(size[0] / self.isoline_voxel_size_value()) + 1,
+                int(size[1] / self.isoline_voxel_size_value()) + 1,
+                1,
+            ],
+            dtype=np.int64,
+        )
+
+        # Calculate the cutoffs for the shells
+        shells_cut_off = self.calculate_cutoffs()
+
+        grid = np.array(
+            generate_voxel_grid(
+                origin,
+                voxel_size,
+                voxel_number,
+                self.aos,
+                mo_coefficients,
+                shells_cut_off,
+            ),
+        )[:, :, 0]
+        self.isoline_voxel_grid.set_grid(grid, origin, voxel_size[:2])
+
+    def set_isolines_hidden(self) -> None:
+        """Set the isolines to hidden."""
+        self.isolines_are_visible = False
+        self.ui.displayIsolinesButton.setText("Display Isolines")
+
+    def set_isolines_visible(self) -> None:
+        """Set the isolines to visible."""
+        self.isolines_are_visible = True
+        self.ui.displayIsolinesButton.setText("Hide Isolines")
+
+    def toggle_isolines(self) -> None:
+        """Toggle the display of the isolines."""
+        self.isolines_are_visible = not self.isolines_are_visible
+        if self.isolines_are_visible:
+            self.ui.displayIsolinesButton.setText("Hide Isolines")
+            self.visualize_isolines()
+            return
+        self.ui.displayIsolinesButton.setText("Display Isolines")
+        self.remove_isolines()
+
+    def visualize_isolines(self) -> None:
+        """Calculate the voxel grid."""
+        assert self.aos is not None
+        assert self.mos is not None
+        self.remove_isolines()
+
+        number_of_iso_values = self.ui.numberLinesSpinBox.value()
+
+        if self.isoline_grid_parameters_changed:
+            self.calculate_isoline_voxelgrid()
+            self.isoline_grid_parameters_changed = False
+
+        grid = self.isoline_voxel_grid.grid
+        origin = self.isoline_voxel_grid.origin
+        voxel_size = self.isoline_voxel_grid.voxel_size
+        voxel_number = self.isoline_voxel_grid.voxel_number
+
+        grid_sum = np.sum(np.abs(grid))
+        zero_approx = 1.0e-14
+        if grid_sum > zero_approx:
```

Comment:
```suggestion
        if grid_sum > np.finfo(float).eps:
```

This would be equal to 2.220446049250313e-16, if that's sufficient